app/src/org/commcare/android/database/user/models/FormRecord.java

```diff
@@ -73,6 +91,12 @@
      * object representation of to reference in the short-term
      */
     public static final String STATUS_JUST_DELETED = "just-deleted";
+    private static final String TAG = FormRecord.class.getName();
+
+    @StringDef({STATUS_UNSTARTED, STATUS_INCOMPLETE, STATUS_COMPLETE, STATUS_UNSENT, STATUS_SAVED, STATUS_QUARANTINED, STATUS_UNINDEXED, STATUS_JUST_DELETED})

```

Comment:
This is a really cool pattern, thanks for sharing it!

Reply:
Yeah. It also saves memory overhead caused by Enums as these annotations are compile time only.